### PR TITLE
Fixes #7802: Make reading/writitng StringTemplate template a seperate service

### PR DESCRIPTION
**Pull request by fanf**
https://www.rudder-project.org/redmine/issues/7802
All feedbacks are implemented, this can be merged

### CODE REVIEW

**Comment by fanf** [comment]
PR updated

**Comment by fanf** [comment]
PR updated

**Comment by fanf** [comment]
Have to add a --stdout option and a fallback case: when no template are given, try to read from stdin.

File: rudder-templates-cli/src/main/scala/com/normation/templates/cli/TemplateCli.scala [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+
+import net.liftweb.common._
+import net.liftweb.json._
+import scopt.OptionParser
+
+/**
+ * The configuration object for our CLI.
+ * The basic process is to take one file in input for the definition of variables, one set of files as template to change.
+ *
+ * By default, the files are generated with the same name in a "out" folder, under the current one.
```
Comment (VinceMacBuche) [inline rudder-templates-cli/src/main/scala/com/normation/templates/cli/TemplateCli.scala]: wromng comment on default value for out

File: rudder-templates-cli/src/main/scala/com/normation/templates/cli/TemplateCli.scala [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+    head("rudder-templates-cli", "3.3.x")
+
+    opt[File]('o', "out") valueName("<file>") action { (x, c) =>
+      c.copy(out = x) } text("output directory for filled template, default is '.'")
+
+    opt[String]("inext") optional() valueName("<input file extension>") action { (x, c) =>
+       c.copy(inputExtension = x) } text("extension of input templates. Default is '.st'")
+
+    opt[String]("outext") optional() valueName("<output file extension>") action { (x, c) =>
+       c.copy(inputExtension = x) } text("extension of templates after processing. Default is '.cf'")
```
Comment (VinceMacBuche) [inline rudder-templates-cli/src/main/scala/com/normation/templates/cli/TemplateCli.scala]: default is ""

Comment (VinceMacBuche) [inline rudder-templates-cli/src/main/scala/com/normation/templates/cli/TemplateCli.scala]: And you modify inoutExtension value !! shout be outputExtension

File: rudder-templates-cli/src/main/scala/com/normation/templates/cli/TemplateCli.scala [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+    //in case of error with args, stop and display usage
+    val config = parser.parse(args, Config()).getOrElse {
+        parser.showUsage
+        System.exit(1)
+        //just for type inference, never reached
+        Config()
+    }
+
+
+    val res = for {
```
Comment (VinceMacBuche) [inline rudder-templates-cli/src/main/scala/com/normation/templates/cli/TemplateCli.scala]: Maybe just need a comment explaining your are filling templates here or just rename res to "filledTemplates"?

**Comment by fanf** [comment]
PR updated

**Comment by VinceMacBuche** [comment]
It's ok and can be merged !

**Pull request merged**